tasks/eva_2121/aggregate_table.py

```diff
@@ -0,0 +1,226 @@
+        elif 'no' not in stay_unchanged:
+            unchanged_taxid.append(taxid)
+        elif source == 'EVA':
+            eva_only.append(taxid)
+        elif source == 'DBSNP':
+            dbsnps_only.append(taxid)
+        else:
+            mixture.append(taxid)
+
+    # Iterate a second time to assign a temp mongodb per taxonomy
+    with open(output_taxonmomy_tsv, 'w') as open_output:
+        headers = ['taxonomy_id', 'scientific_name', 'temp_mongo', 'to_copy', 'number_variants_to_process', 'number_variants']
+        print('\t'.join([str(o) for o in headers]), file=open_output)
+
+        for taxid in data_per_taxid:
+            rows = data_per_taxid[taxid]
+            scientific_name = {row['Scientific Name From Taxid'] for row in rows}.pop()
+            number_variants_to_process = count_variants(rows)
+            taxid_to_tempmongo[taxid] = assign_tempmongo_host_round_robin(number_variants_to_process,
+                                                                          total_number_variant)
+
+            if taxid in unchanged_taxid + only_unmapped_tax_id:
+                to_copy = 'no'
+                temp_mongo = ''
+            else:
+                to_copy = 'yes'
+                temp_mongo = taxid_to_tempmongo[taxid]
+
+            print('\t'.join([
+                str(taxid),
+                scientific_name,
+                temp_mongo,
+                to_copy,
+                str(number_variants_to_process),
+                str(count_variants(rows, False))
+            ]), file=open_output)
+
+    with open(output_assemblies_tsv, 'w') as open_output:
+        headers = ['taxonomy_id', 'scientific_name', 'assembly', 'sources', 'fasta_path', 'report_path', 'temp_mongo',
+                   'to_process', 'number_variants_to_process', 'number_variants']
+        print('\t'.join([str(o) for o in headers]), file=open_output)
+        for taxid, assembly in data_per_taxid_and_assembly:
+            rows = data_per_taxid_and_assembly[(taxid, assembly)]
+            scientific_name = {row['Scientific Name From Taxid'] for row in rows}.pop()
+            number_variants = sum([int(row['Number Of Variants (submitted variants)'].replace(',', '')) for row in rows])
+
+            if taxid in unchanged_taxid + only_unmapped_tax_id or assembly == 'Unmapped':
+                to_process = 'no'
+                fasta, report, temp_mongo = '', '', ''
+            else:
+                to_process = 'yes'
+                fasta, report = retrieve_assembly(scientific_name, assembly, assembly_dir, download_dir)
+                temp_mongo = taxid_to_tempmongo[taxid]
+
+            out = [
+                taxid,
+                scientific_name,
+                assembly,
+                ','.join({row['Source'] for row in rows}),
+                fasta,
+                report,
+                temp_mongo,
+                to_process,
+                number_variants,
+                str(count_variants(rows, False))
```

Comment:
Given the logic in count_variants, won't the result be the same as number_variants?

Reply:
No, they will be the same number when all the variants for will be processed. But will be different when row with unmapped data.